### PR TITLE
Implement Translation and Authentication

### PR DESCRIPTION
**Pull request by dtalton3**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/UserServlet.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            
+        } else {
+            userCredentials.put("valid", new Boolean(false));
+            userCredentials.put("loginUrl", userService.createLoginURL("/index.html"));
+        }
+
+        response.setContentType("application/json");
+        response.getWriter().println(convertToJsonUsingGson(userCredentials));
+  }
+
```
Comment (octopus1000) [inline portfolio/src/main/java/com/google/sps/servlets/UserServlet.java:61]: Remove this?

Comment (dtalton3) [reply portfolio/src/main/java/com/google/sps/servlets/UserServlet.java:61]: Done.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -101,12 +93,15 @@ private String convertToJsonUsingGson(Object list) {
...
   }
 
-  public List<String> loadCommentH() {
+  public List<String> loadCommentH(String lang) {
     List<String> list = new ArrayList<String>();
     Query query = new Query("Comment").addSort("timestamp", SortDirection.ASCENDING);
     PreparedQuery results = datastore.prepare(query);
     for (Entity entity : results.asIterable()) {
-        list.add((String) entity.getProperty("Comment"));
+        Translate translate = TranslateOptions.getDefaultInstance().getService();
```
Comment (octopus1000) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:101]: Nit. Move this statement out side of iteration so you don't have to get translation service in every iteration.

File: portfolio/src/main/webapp/index.html [hunk 3]
```diff
@@ -95,20 +95,26 @@ <h1>
...
+        <div id=login></div>
         <div id="commentsection">
-                <p>Type a comment:</p>
-                <input type="text" name="commentpost" value="" id="textfield">
-                    <button onclick="showComments()">Post.</button>
-                <br/><br/>
-               </input>
         </div>
+        <script>
+                authenticateUser();
```
Comment (octopus1000) [inline portfolio/src/main/webapp/index.html:116]: Move <script> below </body>

<script>
authenticateUser()
loadComments()
</script>

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -29,7 +29,7 @@ function funFactGenerator() {
 
   // Add it to the page.
   const factGetter = document.getElementById('factgetter');
-  factGetter.innerText = fact;
+  factGetter.innerText = fact;it
```
Comment (octopus1000) [inline portfolio/src/main/webapp/script.js:32]: it?

**Review by octopus1000 — APPROVED** [review]

**Pull request merged**